### DISCUSSION TITLE
FIP Editor Membership Expectation

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/FIPs#960] [Enhancements - Organizational]
> Disclaimer: this discussion does NOT intend to propose any changes to what's in FIP-0001 wrt FIP Editors. 

I would like to open a discussion to discuss and hopefully experiment with some mechanism to apply FIP editor expectation, with the goal of improving the experience for Filecoin community members to enage in the FIP process.  

Some basis this post is based on
- [FIP Editor Responsibililties/Do-es/Not Do-es](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md#fip-editor-responsibilities).
- @TippyFlitsUK joined FIP editor group recently, and we have in total 8 FIP editors.
- FIP editing today isi 100% public good work. Other than @kaitlin-beegle, who is hired by FF as the Governance/FIP PM, most of the rest FIP editors are voluntarily donating their personal time to support the FIP editing needs. 

**So..what's the issue today?**
Most of the FIP editors have many other responsibilities they have to fulfill during their paid working hours, incluing but not limited to: protocol R&D, develop implementations, support various of ecosystem functions, lead/build/support network teams & etc. For the lack of better words - they are busy 🐝 😄 However, even without any direct incentive, most FIP editors are trying their best to take on the responsibilities in their own time so to support progressing Filecoin improvement proposals. 
That being said, speaking from my own experience:
- when other higher impact work lines up (to the Filecoin project and/or individual personally),  it is easy to depriotize FIP editing work, and that ⚠️  leads to slow turn around time on FIP PR engagement, which may furthur slow down FIP development and potentially lower the interest from FIP authors to participate in the process. Sometime even worse - may impact how fast a good protocol proposal may be shipped to the network.
- when few FIP editors are picking up the work, that leads to even heaiver workload to the ones that are engaging, which either takes their time away from other important work they can contribute to Filecoin, or causing a burn out on this line of work ->  wretched cycle to the point above. 

**Why do we need to fix this?**
To evolve Filecoin fastly, we want to expand and scale the Filecoin protocol developer pool as much as possible, it is important for those who are interested in contributing Filecoin protocol to have a good OSS developement experience by getting the engagement, support, mentorship and feedback in a timely manner. 

 
**How can we fix this?**
Note that I have **a lot of** respect to my fellow editors  and like mentioned, I know for a fact everyone is doing their best 💙. The goal here is trying to maintain a minial standard and expectation of the FIP editor group for the community, and hopefully to help the group to sustain.  

There may be many ways to mitigate this gap, here i want to propose an easy-to-implement approach by setting minimal engagement expectation for FIP editor membership. Even tho most of the FIP editors don't have direct incentives and are contributing to this public effort, however I do think there are some implict "privillage" being an FIP editor like social capitial/reputation and learning and collaberation opportunities with FIP authors. Or more generally, imho following a simple prinicple, if taking a role/title == committing to the responsibilities -> do the work to fulfill the responsibility. If one can't fulfill the role that is totally fine, and it is better for the Filecoin project to be honest with the group/Governace TPMs, so they can recruite others to join the effort. 

I did some quick data collecting for FIP activity happened in the past 3 quarters to understand the workloads.
<img width="1367" alt="image" src="https://github.com/filecoin-project/FIPs/assets/42981373/b3c13408-5c02-48d9-998a-aeb62ae37580">

I think if editors decouple the editorial review and peer review, **the expected avg time commitment is 0-2 hr/week, and the avg time commitment might be even lower if 8 editors share workloads more evently.** I explictly meant to callout decouping the reviews as a thorough peer reviews could take significantly longer, i.e: editorial review on DDO may take me ~1 hr but a peer review may take me a couple 2hr-sessions to complete.

So my proposal is FIP editors can always voluntarily resign if they can no longer committing their time to this public good work, and **anyone can **_suggest_**  an FIP editor to be removed if they review less than 70% of the expected avg PR # for 2 consecutive months.** 

> e:g: if the total amount of PR closed was _10_, a total of _10*2=20_ reviews are needed across 8 edtitors. avg review is _20/8_~=3, if an fip editor reviewed less than 3 PRs that month, that month is cosidered as less than 70% of the expected PR reviewed.

I believe if the FIP editors work tgt collaberatively, the time commitment ask is quite low based on theh current FIP volume, and I am hoping this small push on expectation can help improve the overall FIP PR turn around t ime for our community. 

**CTA**
- @filecoin-project/fips-editors Do you agree with the motivation? And what's your thought on the initial proposal to experiment with, and can iterate accordingly as needed?
  - especially @momack2 @arajasek @raulk, you folks have relatively lower engagement on FIP editing work in the past couple month, do you want to continue contribute as an FIP editor (needlessly to say, ths is only one of the many ways you are contributing to Filecoin)?
- @kaitlin-beegle @luckyparadise to 🤔 What's the minial FIP editor # needed? If we cant sustain that need from public good contribution, or FIP activity increases, how can we expand and/or scale FIP editor group to meet the community needs?

**Comment by The-Wayvy** [comment]
I suggest Molly be removed

**Reply by jennijuju** [reply]
What engagement expectations/mechanism are you basing on?
Btw-  I think we should only execute some rules once it's established, I'm hoping editors and governance TPM can align on an initial minimal engagement expectation soon (if they support this proposal).

**Comment by luckyparadise** [comment]
Thanks, @jennijuju for starting this discussion. I agree mostly with your suggestions. 

Both Core Devs and FIP Editors will need some level of contribution benchmarking to guide continued membership as either Core Dev or FIP Editor (in this case). There is a need to expand the README for FIP Editors to reflect new changes to ensure the effectiveness of the group:

- We need a FIP Editor Handbook separate from what is currently referenced in the present FIP0001. This should be a longer document scoping out in entirety what it means to be a FIP Editor, minimum commitment and responsibility, steps to join as a FIP Editor, steps to remove an Editor or voluntarily resign. 
- There should be clearly defined steps to take where FIP Editors do not meet these minimum commitments and how these are triggered (ie voting to remove an Editor).
- At the moment, there is no opposition to opening up the composition of FIP Editors as per - [here](https://github.com/filecoin-project/FIPs/discussions/783). The FIP Editor Handbook will contain a lightweight rules-base on a meritocratic approach to including new FIP Editors. This means that anyone can be nominated to be a FIP Editor, provided there is no objection from any FIP Editor. 
- Internally I know @kaitlin-beegle is doing some design work on incentivizing FIP Editor review work, this is still at a very early stage but it is something we are taking very seriously.  

In summary, I am in support of getting clear contribution expectations from all FIP Editors, in addition, there should be a distinct document that describes in detail what this 'contribution expectation' means.  

 Ultimately, we need all the FIP Editors to agree to the need for these changes and publicly express interest and impetus to back these changes.

**Reply by kaitlin-beegle** [reply]
+1 on the FIP Editor Handbook.  We can pretty quickly build out a space for this to live.  

@luckyparadise want to discuss this during next week's check-in?

**Comment by anorth** [comment]
I concur that having people in the FIP editor role who are not doing the work is counter-productive. It is a role of service, not authority. People who do not wish to provide this service should resign or be removed. I have little sympathy for anyone wanting to maintain the title as vanity. We have removed inactive FIP editors in the past. I think we should do so again and be clear if we need to recruit more.

**Reply by arajasek** [reply]
Strong agree, this is not a vanity title, it's a responsibility.

**Comment by rvagg** [comment]
Something that I've seen work well in other ecosystems is just having a bot to do the work of putting up a removal PR after a certain amount of inactivity. It means that someone doesn't have to be the bad-guy and the rules are enshrined in code, but as a PR the flexibility for discussion remains.

**Reply by kaitlin-beegle** [reply]
Not a bad suggestion, but I don't think that inactivity on certain FIPs is the issue.  Big proposals can take a long time, and I don't think speed _per se_ is the right metric to prioritize.

**Comment by kaitlin-beegle** [comment]
Thanks for sharing your thoughts, @jennijuju.  

In general, my concern at the moment is the typical flow of resources, participation, and engagement in this ecosystem is changing, and this change is deeply affecting processes that were already strained.  And perhaps for the better! But there remains a question of how best to adapt. 

First, core teams and contributors seem to be operating in an increasingly decentralized fashion, but this rate of change may be too great for our processes to continue to be governed more by norms than by strict rules. 
     - _For this reason, I want to second @luckyparadise's suggestion to create a more formal handbook for FIP Editors._ 
     - _This handbook should not only summarize expectations, but should more strictly lay out governing procedures for who is, remains, and can become a FIP Editor._ 

Second, as @jennijuju alluded, broader network changes mean that the Filecoin Foundation should reconsider how it is incentivizing and coordinating governance activities that were previously managed by volunteers.  This is a broader shift, but it is worth calling out the careful balance between paying fairly for time -bounded, skilled work, and the open source ethos of volunteerism and free contribution.  For FIP Editors at least, I increasingly think that the seesaw is tipped towards the former. 

For now, I appreciate @jennijuju summarizing recent contributions.  **I don't think we need to set a quota for committed work, but I think it's reasonable for Aayush, Raul, and Molly to be removed as editors.** If they want to resume their roles, they can request to do so in the future. 

And in the meantime, it seems that I can likewise contribute more.

**Reply by arajasek** [reply]
Yeah, I think this is reasonable, and agree that I shouldn't be a FIP editor. Folks can always rejoin in the future if they want to engage that way.

**Comment by jsoares** [comment]
I agree with this _in principle_ but there's nuance. As @jennijuju writes, this is mainly public goods work that most people aren't being compensated for. That's fine, but it also means that they don't always get to prioritise it over their other responsibilities, and therefore their involvement may be spiky. I'm definitely guilty of this myself, in that sometimes I can dedicate full days to reviewing FIPs and other times I go for weeks without doing much. When major network-wide events are taking place (e.g. upgrades) and given the current make-up of the group, activity can drop across the board. This will only get worse as people disperse into different organisations.

I can also relate to the frustration of having FIPs pending for weeks because of slow or limited editor engagement. Having a better distribution of work would help, but so would just enlisting more editors. And, of course, just the number of reviews doesn't tell the whole story, as the time to review is a more relevant indicator; Little's Law and all, latency will increase dramatically if we operate close to capacity. Instead, we should aim for overcapacity, so that the spikes would be less correlated and there would always be capacity available. That comes with challenges too because we need to maintain consistent standards and that gets superlinearly harder with group size.

I'm wary of imposing too strict restrictions on the role that could lead well-intentioned people to leave (because they couldn't find the time for a month or two) and newcomers to have second thoughts about joining (by causing them to question their ability to meet expectations). Having explicit incentives would make such strict requirements more palatable but the bigger question is how do we find and encourage the right people to volunteer.

**Reply by jennijuju** [reply]
> I'm wary of imposing too strict restrictions on the role that could lead well-intentioned people to leave (because they couldn't find the time for a month or two) and newcomers to have second thoughts about joining (by causing them to question their ability to meet expectations). Having explicit incentives would make such strict requirements more palatable but the bigger question is how do we find and encourage the right people to volunteer.

agree with everything that’s said, especially on “no imposing too strict restriction” , after all this is public good and voluntarily work, and all contribution even small, counts!

**Comment by Fatman13** [comment]
> So..what's the issue today?

i think another issue today is that FIP editors are not decentralized enough as most of the editors are coming from one single DevCo, which makes the flavor of FIPs a bit monolithic.

**Comment by raulk** [comment]
Happy to be removed if that's the consensus amongst fellow FIP editors! But here are my considerations:

1. I think the editorial bottlenecks would be better addressed by [_expanding_ this group](https://github.com/filecoin-project/FIPs/discussions/783) (discussion stale since 2023) rather than contracting it, at least at this point in time. AFAIK, governance FIPs don't impose any limits on editor count, so it's unclear how pruning this group right now solves the productivity/latency problems.

2. While I haven't been able to engage at a steady rate —nor with immediacy— my contributions have been non-zero. I can't compete in availability and capacity with folks who are salaried to work directly on the core protocol (e.g. FilOz) and/or governance (e.g. FF), and whose daily workplace is, quite literally, this repo. In fact, establishing the same bar for volunteers from adjacent orgs, and for salaried groups, is unjust.

3. I always viewed my editor membership as "best effort". I uphold that self-imposed expectation, but as some folks may know here, since Aug'23 I've been swamped onboarding to, launching, and leading, other initiatives that directly contribute to Filecoin's impact (IPC). Unfortunately that has eaten away at my availability for volunteer work, and my editorial responsibilities have taken a hit. However, I do engage opportunistically and serendipitously at times (e.g. when a keyword catches my attention in my inbox), and I had planned on continue doing so.

4. The timesheet-based regime introduced above is flawed, and rather antithetical to volunteer/OSS arrangements (that has yet not reached a productive equilibrium). Rather than imposing time expectations ("0-2hr per week") and shrinking membership to individuals who can meet those expectations, I'd rather onboard new batches of editors from ecosystem teams favouring a mindset of abundance, valuing diversity, accepting opportunistic participation, and investing in resilience. There are tens of Filecoin community members that have demonstrated discernment, impartiality, productive+resolutive discourse skills, and reasonable protocol knowledge (not required but helpful for more holistic assessments) throughout FIP discussions. IMO, the Filecoin spec community would become more vibrant and expeditious if we expanded this group to 10-20 editors from a variety of affiliations and diverse perspectives, without imposing time expectations on each, but rather keeping it fluid, interesting and fun, rather than sucking the life out of it with metrics people need to meet.

For now, I've gone ahead and removed myself from @filecoin-project/fips-editors, and have approved https://github.com/filecoin-project/FIPs/pull/991. I'll keep an eye on this thread in case people want to comment on my views.

**Reply by The-Wayvy** [reply]
Based
